### ISSUE TITLE
Passwords cannot be saved into keychain on macOS with QGIS 3.16

### ISSUE THREAD

**Issue by m-kuhn** [qgis/QGIS#40541]
Since upgrading to QGIS 3.16, the keychain on macos is no longer working.
It used to work on the same system with QGIS 3.14.

![Bild 10 12 20 um 09 43](https://user-images.githubusercontent.com/588407/101748138-74aa7680-3acc-11eb-849b-81f8338507b5.jpg)
![Bild 10 12 20 um 09 44](https://user-images.githubusercontent.com/588407/101748147-77a56700-3acc-11eb-8e41-1f90f5530928.jpg)

**Comment by juneeighteen** [comment]
👁️ Also seeing the same issue. 
macOS Big Sur 11.1
QGIS 3.16

**Comment by 3nids** [comment]
I don't have any issue a self-compiled master on Big Sur.
![image](https://user-images.githubusercontent.com/127259/105709514-1c152880-5f16-11eb-8de7-77dfbeb18c77.png)

**Comment by PeterPetrik** [comment]
3.14 and 3.16+ uses completely different packaging mechanics. So if any, the problem would be with packaging rather than with the QGIS code itself

**Comment by PeterPetrik** [comment]
Hi @juneeighteen and @m-kuhn can you try with https://qgis.org/downloads/macos/nightly/qgis_nightly_master_20210214_122747.dmg ? 

I have updated qt keychain to version 0.12.0, let see if the bug is still there?

**Comment by juneeighteen** [comment]
Sorry I didn't get back sooner. I upgraded to3.16.3-Hannover and the issue resolved itself.

**Comment by PeterPetrik** [comment]
great, any change to test with the nightly?

**Comment by soester** [comment]
for me the problem still persists in QGIS 3.16

------

QGIS version | 3.16.3-Hannover | QGIS code revision | 3a90ea3afd
-- | -- | -- | --
Compiled against Qt | 5.14.2 | Running against Qt | 5.14.2
Compiled against GDAL/OGR | 3.1.2 | Running against GDAL/OGR | 3.1.2
Compiled against GEOS | 3.8.1-CAPI-1.13.3 | Running against GEOS | 3.8.1-CAPI-1.13.3
Compiled against SQLite | 3.31.1 | Running against SQLite | 3.31.1
PostgreSQL Client Version | 12.3 | SpatiaLite Version | 4.3.0a
QWT Version | 6.1.4 | QScintilla2 Version | 2.11.4
Compiled against PROJ | 6.3.2 | Running against PROJ | Rel. 6.3.2, May 1st, 2020
OS Version | macOS 10.15
Active python plugins | QgisModelBaker; trackable_project_files; changeDataSource; quick_map_services; SRTM-Downloader; qfieldsync; pg_history_viewer; swiss_locator; processing; db_manager

------

MacOS 10.15.7

**Comment by PeterPetrik** [comment]
@soester can you please test with latest nightly build?

**Comment by m-kuhn** [comment]
I'm not close to a computer where I could check these days, I'm sorry. Thanks for taking care @PeterPetrik and for testing @juneeighteen and @soester

**Comment by soester** [comment]
Also doesn't work on nightly..
QGIS version | 3.17.0-Master | QGIS code revision | 4fc02733cb

I now also noticed this banner:
<img width="789" alt="Bildschirmfoto 2021-02-18 um 12 10 08" src="https://user-images.githubusercontent.com/28543978/108349149-f0b6ee00-71e2-11eb-82bf-851ab55a1b53.png">

**Comment by authalic** [comment]
Also seeing this on QGIS 3.18.2 
macOS 11.3
with data store in PostgreSQL 13.2.2

**Comment by PeterPetrik** [comment]
@authalic can you please share the step guide how to replicate the problem (with screenshot?) thanks

**Comment by Rikuoja** [comment]
I am seeing the same when opening a project in
PostgreSQL 13.2
QGIS 3.18.2
Mac OS 11.2.3
MacBook Air (Apple M1 ARM processor)

<img width="456" alt="Näyttökuva 2021-5-6 kello 12 41 10" src="https://user-images.githubusercontent.com/9367712/117277469-95eb7500-ae68-11eb-9bc3-10589e1adc79.png">

1. Opening this project results in 

<img width="1440" alt="image" src="https://user-images.githubusercontent.com/9367712/117277671-c0d5c900-ae68-11eb-8bd6-4d9b9bac1334.png">

2. Typing in the password in the prompt again results in

<img width="1142" alt="Näyttökuva 2021-5-6 kello 12 44 42" src="https://user-images.githubusercontent.com/9367712/117277844-e4990f00-ae68-11eb-81a4-6d785f4c9de4.png">

which is the Finnish text of the German WARNING above.

**Comment by Rikuoja** [comment]
The password helper debug log prints
```2021-05-12T12:08:50     WARNING    Avataan Keychain operaatiolle READ…
2021-05-12T12:08:50     WARNING    Error in Keychain: Retrieving password from your Keychain failed: OS X Keychain error (OSStatus -25293).
2021-05-12T12:08:57     WARNING    Avataan Keychain operaatiolle WRITE…
2021-05-12T12:08:57     WARNING    Error in Keychain: Storing password in your Keychain failed: OS X Keychain error (OSStatus -25293).```

**Comment by PeterPetrik** [comment]
Hi, can you please try:

1) enable password debug log and see if there are any useful entries?
2) try to uncheck "integrate your master password with keychain" and try if it works? 
3) enable it back and "clear the master password from keychain" and reenter it? check if works?
4) try to clear various cache options and see if it helps ? 

<img width="1330" alt="Screenshot 2021-06-14 at 13 46 04" src="https://user-images.githubusercontent.com/804608/121887461-095d8d80-cd17-11eb-921c-247cbe9338ab.png">

**Comment by soester** [comment]
@PeterPetrik 
i tested your suggestions above and get the same warnings:
2021-06-24T16:45:19     WARNING    Opening Keychain for READ…
2021-06-24T16:45:19     WARNING    Error in Keychain: Retrieving password from your Keychain failed: OS X Keychain error (OSStatus -25293).
2021-06-24T16:45:29     WARNING    Opening Keychain for WRITE…
2021-06-24T16:45:29     WARNING    Error in Keychain: Storing password in your Keychain failed: OS X Keychain error (OSStatus -25293).
2021-06-24T16:47:21     WARNING    Opening Keychain for WRITE…
2021-06-24T16:47:21     WARNING    Error in Keychain: Storing password in your Keychain failed: OS X Keychain error (OSStatus -25293).
2021-06-24T16:48:21     WARNING    Opening Keychain for DELETE…
2021-06-24T16:48:21     WARNING    Error in Keychain: Delete password failed: OS X Keychain error (OSStatus -25293).
2021-06-24T16:50:00     WARNING    Opening Keychain for WRITE…
2021-06-24T16:50:00     WARNING    Error in Keychain: Storing password in your Keychain failed: OS X Keychain error (OSStatus -25293).
2021-06-24T16:58:58     WARNING    Opening Keychain for WRITE…
2021-06-24T16:58:58     WARNING    Error in Keychain: Storing password in your Keychain failed: OS X Keychain error (OSStatus -25293).

**Comment by soester** [comment]
I disabled 'Integrate master password in Keychain' with QGIS 3.18 as it still didn't work.
Now i enabled it again with 3.20 to test. And it work's :)

However the issue persists  for QGIS  3.16.8.

**Comment by soester** [comment]
@gioman 

> **However the issue persists for QGIS 3.16.8.**
Shouldn't we leave this issue open?
Or did you find a solution to fix this in 3.16?

**Comment by gioman** [comment]
> Shouldn't we leave this issue open?
> Or did you find a solution to fix this in 3.16?

@soester if it is fixed... is fixed. What eventually need to be asked is to backport the fix (assuming is possible).
